docs/upgrade/upgrade-instructions-for-entities-simplification.md

```diff
@@ -0,0 +1,166 @@
++ CustomerFacade::__construct(EntityManagerInterface $em, UserRepository $userRepository, CustomerDataFactoryInterface $customerDataFactory, CustomerMailFacade $customerMailFacade, BillingAddressFactoryInterface $billingAddressFactory, DeliveryAddressFactoryInterface $deliveryAddressFactory, BillingAddressDataFactoryInterface $billingAddressDataFactory, UserFactoryInterface $userFactory, CustomerPasswordFacade $customerPasswordFacade)
+```
+```diff
+- User::edit(UserData $userData, EncoderFactoryInterface $encoderFactory)
++ User::edit(UserData $userData)
+```
+```diff
+- User::__construct(EntityNameResolver $entityNameResolver, EncoderFactoryInterface $encoderFactory)
++ User::__construct(EntityNameResolver $entityNameResolver, CustomerPasswordFacade $customerPasswordFacade)
+```
+```diff
+- User::resetPassword(HashGenerator $hashGenerator)
++ User::resetPassword(string $resetPasswordHash)
+```
+```diff
+- Administrator::edit(AdministratorData $administratorData, EncoderFactoryInterface $encoderFactory, ?self $administratorByUserName)
++ Administrator::edit(AdministratorData $administratorData)
+```
+
+Additional information:
+- protected constant `DEFAULT_PRODUCT_QUANTITY` has been moved from `Order` to `OrderItemFacade`
+- `InvalidQuantifiedProductException` has been removed
+- there is a new method `ProductCategoryDomainFactoryInterface::createMultiple()` that needs to be implemented
+    - you can get inspired by the [implementation in the framework package](https://github.com/shopsys/shopsys/blob/6f77d54098e584a3a762d65addb399e4366c33b6/packages/framework/src/Model/Product/ProductCategoryDomainFactory.php#L44)
+- `Product::edit()` no longer schedules product for price recalculation
+    - you need to use `ProductFacade::edit()`
+- `CustomerPasswordFacade::setNewPassword()` is now strictly type hinted
+- protected constant `RESET_PASSWORD_HASH_LENGTH` has been moved from `User` to `CustomerPasswordFacade` and made public
+- use `AdministratorFacade::edit` instead of `Administrator::edit()` if you want to keep the checking of the admin's username uniqueness
+
+Tests:
+- in many tests, methods that changed their signatures are called
+    - running `php phing phpstan` should help you to discover the problems
+- remove `Tests\ShopBundle\Functional\Model\Cart\CartTest::testMergeWithCartReturnsCartWithSummedProducts()`  
+    - add new `Tests\ShopBundle\Functional\Model\Cart\CartMigrationFacadeTest` class, you can copy-paste it from [Github](https://github.com/shopsys/shopsys/blob/1c62e6c1f31ba8b5437154f2ae419f3238083ce4/project-base/tests/ShopBundle/Functional/Model/Cart/CartMigrationFacadeTest.php)
+- move tests from `Tests\ShopBundle\Functional\Model\Cart\CartTest` to `Tests\ShopBundle\Functional\Model\Cart\CartFacadeTest`
+and fix them appropriately (you can copy paste them from [Github](https://github.com/shopsys/shopsys/blob/24ae0ab48f9059af5da81449e83d5d5b50d38d98/project-base/tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php))
```

Comment:
what about ? 
```suggestion
and fix them appropriately (you can copy paste them from [shopsys/ptoject-base](https://github.com/shopsys/project-base/blob/v8.0.0/tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php))
```
if it is OK, please update it for all those occurrences.

Reply:
fixed everywhere :wink: